### PR TITLE
Add disableDates as an option on the Date Picker

### PR DESCRIPTION
**Pull request by ehubbell**
Optional param that let's you supply an array of dates that you'd like disable on the date picker. A good use case might be a business that's already booked on that day.

### CODE REVIEW

**Comment by mydea** [comment]
Thank you for the PR!
I went through it and added some comments. If you could resolve them, I'd be happy to help. If you need any more guidance, please let me know.

Some general things in addition to the comments:

- Please make sure that everywhere where you use `disabledDates` you fall back to an empty array, if not specified. e.g. `let disabledDates = get(this, 'disabledDates') || [];`
- Please remove the package-lock.json file, as this project uses yarn, and this is otherwise a bit confusing ;)
- Please ensure that all current tests pass. There are especially some linting errors.
- Ideally, please also add one test for the date-picker-month container with some disabledDates set

**Comment by ehubbell** [comment]
Pretty sure I got everything fixed up as asked. First time writing QUnit tests so it was nice learning and getting some practice. I may end up submitting a couple other pull-requests in the future so I'll keep these constructs in mind. Lemme know any questions!

File: addon/components/date-picker.js [hunk 1]
```diff
@@ -36,6 +36,17 @@ export default Component.extend({
...
+  /**
+   * An array of optional dates to disable for this date picker.
+   * These dates will not be selectable.
+   *
+   * @attribute disableDates
+   * @type [Date]
+   * @optional
+   * @public
+   */
+  disableDates: null,
```
Comment (mydea) [inline addon/components/date-picker.js]: Also rename this one to `disabledDates`.

File: addon/components/date-picker-month.js [hunk 2]
```diff
@@ -269,16 +300,37 @@ export default Component.extend({
...
+  /**
+   * Check if a date is disabled.
+   * This checks if the date is inside of minDate/maxDate.
+   *
+   * @method _dayIsDisabled
+   * @param {Date} day The date to check
+   * @return {Boolean}
+   * @private
+   */
+  _dayIsDisabled(day) {
```
[the remark this answers is not in the document]
Comment (mydea) [reply addon/components/date-picker-month.js]: I think we can overall simplify this function to:

```js
_dateIsDisabled(day) {
  let disabledDates = get(this, 'disabledDates') || [];
  return !!disabledDates.find((date) => date.isSame(day, 'day'));
}
```

**Comment by mydea** [comment]
The changes look good!

A few things to go (see the comments in the code) here: https://github.com/mydea/ember-date-components/pull/15/files!

Thanks for the effort :)

**Comment by ehubbell** [comment]
Sorry, I'm confused. I don't see any comments in the code at that URL.  Would you mind pasting them here?

**Comment by mydea** [comment]
Sorry, forgot to actually submit the review ;)

**Comment by mydea** [comment]
I added one last comment - that should be easy to fix :) Then I'll merge it!

File: addon/components/date-picker-month.js [hunk 3]
```diff
@@ -221,14 +203,14 @@ export default Component.extend({
    * @readOnly
    * @private
    */
-  daysInMonth: computed('_daysInMonth', '_disableDates', '_minDate', '_maxDate', 'selectedDates.[]', function() {
+  daysInMonth: computed('_daysInMonth', '_minDate', '_maxDate', 'selectedDates.[]', function() {
```
[the remark this answers is not in the document]
Comment (ehubbell) [reply addon/components/date-picker-month.js]: No worries. Just submitted another pr.

**Comment by ehubbell** [comment]
Hey - just an fyi it looks like you may need to "release" these changes in order to make them accessible via NPM etc.

**Comment by mydea** [comment]
Yeah, I'll probably only get around to it early next week. Need to write documentation for it first!

**Comment by ehubbell** [comment]
Okay cool. Thanks for the update.

**Pull request merged**